src/main/fish/payara/server/PayaraInstanceController.ts

```diff
@@ -20,44 +20,144 @@
-import { QuickPickItem, CancellationToken } from 'vscode';
+import { PayaraServerInstance, InstanceState } from './PayaraServerInstance';
+import { JvmConfigReader } from './start/JvmConfigReader';
+import { JDKVersion } from './start/JDKVersion';
+import { QuickPickItem, CancellationToken, Uri } from 'vscode';
+import { JvmOption } from './start/JvmOption';
+import { StringUtils } from './tooling/utils/StringUtils';
+import { ServerUtils } from './tooling/utils/ServerUtils';
+import { JavaUtils } from './tooling/utils/JavaUtils';
+import { StartTask } from './start/StartTask';
+import { ChildProcess } from 'child_process';
+import { RestEndpoints } from './endpoints/RestEndpoints';
+import { URL } from 'url';
 
 export class PayaraInstanceController {
 
-    private outputChannel: vscode.OutputChannel;
-
     constructor(private instanceProvider: PayaraInstanceProvider, private extensionPath: string) {
-        this.outputChannel = vscode.window.createOutputChannel('payara');
         this.init();
     }
 
     private async init(): Promise<void> {
-        let instances = this.instanceProvider.readServerConfig();
+        let instances: any = this.instanceProvider.readServerConfig();
         instances.forEach((instance: any) => {
-            let payaraServerInstance: PayaraServerInstance = new PayaraServerInstance(
+            let payaraServer: PayaraServerInstance = new PayaraServerInstance(
                 instance.name, instance.path, instance.domainName
             );
-            this.instanceProvider.addServer(payaraServerInstance);
+            this.instanceProvider.addServer(payaraServer);
+            payaraServer.checkAliveStatusUsingJPS(() => {
+                payaraServer.getOutputChannel().show(false);
+                payaraServer.connectOutput();
+                payaraServer.setStarted(true);
+                this.refreshServerList();
+            });
         });
         this.refreshServerList();
     }
 
     public async addServer(): Promise<void> {
         ui.MultiStepInput.run(
-            input => this.selectServer(input, {
-            })
+            input => this.selectServer(input,
+                {},
+                payaraServer => {
+                    this.instanceProvider.addServer(payaraServer);
+                    this.refreshServerList();
+                    payaraServer.checkAliveStatusUsingJPS(() => {
+                        payaraServer.setStarted(true);
+                        this.refreshServerList();
+                        payaraServer.connectOutput();
+                    });
+                })
         );
     }
 
-    public async removeServer(payaraServer: PayaraServerInstance): Promise<void> {
-        this.instanceProvider.removeServer(payaraServer);
-        this.refreshServerList();
+    public async startServer(payaraServer: PayaraServerInstance, debug: boolean): Promise<void> {
+        let process: ChildProcess = new StartTask().startServer(payaraServer, debug);
```

Comment:
If you start the server twice it will appear to be not running but stay running

Reply:
Both validations in function and condition to the menu added.

State Running:
![image](https://user-images.githubusercontent.com/15934072/73338459-a2b00f00-429c-11ea-8988-2d9a58f088ce.png)
State Stopped:
![image](https://user-images.githubusercontent.com/15934072/73338482-b3608500-429c-11ea-9e08-a06abb47f67c.png)
State Loading:
![image](https://user-images.githubusercontent.com/15934072/73338526-ce32f980-429c-11ea-8488-e26d6bb6f16c.png)